geode-dunit/src/distributedTest/java/org/apache/geode/test/dunit/rules/tests/DistributedMapTest.java

```diff
@@ -491,4 +495,162 @@ public void getReturnsSameValueInEveryVmAsPutInOtherVm() {
+    public DistributedMap<Object, Object> distributedMap = new DistributedMap<>();
+
+    @Before
+    public void setUp() {
+      getController().invoke(() -> {
+        map.set(distributedMap.map());
+        distributedMap.put("key1", "value1");
+      });
+    }
+
+    @Test
+    public void distributedMapIsAccessibleInEveryVm() {
+      for (VM vm : asList(getVM(0), getVM(1), getVM(2), getVM(3), getController())) {
+        vm.invoke(() -> {
+          assertThat(distributedMap.map()).isSameAs(map.get());
+          assertThat(distributedMap.get("key1")).isEqualTo("value1");
+        });
+      }
+    }
+  }
+
+  public static class HasTwoDistributedMaps implements Serializable {
+
+    private static final AtomicReference<Map<Object, Object>> map1 = new AtomicReference<>();
+    private static final AtomicReference<Map<Object, Object>> map2 = new AtomicReference<>();
+
+    @Rule
+    public DistributedMap<Object, Object> distributedMap1 = new DistributedMap<>();
+    @Rule
+    public DistributedMap<Object, Object> distributedMap2 = new DistributedMap<>();
+
+    @Before
+    public void setUp() {
+      getController().invoke(() -> {
+        map1.set(distributedMap1.map());
+        distributedMap1.put("key1", "value1");
+
+        map2.set(distributedMap2.map());
+        distributedMap2.put("key2", "value2");
+      });
+    }
+
+    @Test
+    public void twoDistributedMapsAreAccessibleInEveryVm() {
+      for (VM vm : asList(getVM(0), getVM(1), getVM(2), getVM(3), getController())) {
+        vm.invoke(() -> {
+          assertThat(distributedMap1.map()).isSameAs(map1.get());
+          assertThat(distributedMap1.get("key1")).isEqualTo("value1");
+          assertThat(distributedMap1.get("key2")).isNull();
+
+          assertThat(distributedMap2.map()).isSameAs(map2.get());
+          assertThat(distributedMap2.get("key1")).isNull();
+          assertThat(distributedMap2.get("key2")).isEqualTo("value2");
+        });
+      }
+    }
+  }
+
+  public static class HasManyDistributedMaps implements Serializable {
+
+    private static final AtomicReference<Map<Object, Object>> map1 = new AtomicReference<>();
+    private static final AtomicReference<Map<Object, Object>> map2 = new AtomicReference<>();
+    private static final AtomicReference<Map<Object, Object>> map3 = new AtomicReference<>();
+
+    @Rule
+    public DistributedMap<Object, Object> distributedMap1 = new DistributedMap<>();
+    @Rule
+    public DistributedMap<Object, Object> distributedMap2 = new DistributedMap<>();
+    @Rule
+    public DistributedMap<Object, Object> distributedMap3 = new DistributedMap<>();
+
+    @Before
+    public void setUp() {
+      getController().invoke(() -> {
+        map1.set(distributedMap1.map());
+        distributedMap1.put("key1", "value1");
+
+        map2.set(distributedMap2.map());
+        distributedMap2.put("key2", "value2");
+
+        map3.set(distributedMap3.map());
+        distributedMap3.put("key3", "value3");
```

Comment:
Might it be worth extracting the hardcoded string keys/values in this test to constants?